programs/svm-spoke/src/instructions/bundle.rs

```diff
@@ -100,6 +103,7 @@ impl RelayerRefundLeaf {
 
 pub fn execute_relayer_refund_leaf<'c, 'info>(
     ctx: Context<'_, '_, 'c, 'info, ExecuteRelayerRefundLeaf<'info>>,
+    deferred_refunds: bool,
```

Comment:
so now the caller of this method chooses to defer the refunds, or not. in theory this lets people front run the executor bot who would be by default (happy path) setting this to false, and then they execute it setting it to true, forcing relayers to get the deferred refund.

Is there a way to programmatically detect this? my guess is that it will be difficult as we'd basically need to try `distribute_relayer_refunds` and only if that fails then  `accrue_relayer_refunds`, which is not really a possible flow as by that point we might have sent _some_ tokens already. this means that we'd somehow need to know before hand that the whole bundle is executable via `distribute_relayer_refunds` before sending any assets.

I guess this is why you structured it like this?

Reply:
Also in the prior implementation the caller implicitly could chose defer refunds by passing all claim accounts. But I agree, this still might be an issue and we could require that the caller always passed all correct mutable ATAs (even if closed or blacklisted) and in the happy path try to transfer all of them. Only if any transfer fails we require that caller had also passed additional claim accounts and defer the claims only for those refunds that did not go through.
Initially I was bit hesitant to do this as that could require doubling the remaining accounts, but its possible to avoid this in the happy path and we are using ALTs anyway. So I will try to do this.

Reply:
ok cool. otherwise I think the way you have it now is ok, btw, it's just a path we need to know about.

Reply:
Unfortunately, I overlooked the fact that CPIs in Solana halt tx execution immediately, so its not really possible to handle failed transfers by accruing the claim and proceeding to the next refund. So will need to leave this as 2 separate execute relayer refund flavors: one with all ATAs and another with all claim accounts.

As for the original concern, there is not a really good way to protect against frontrunning. Though, we can still claim relayer refunds automatically for all relayers if we detect this.